syntax = "proto3";
option go_package = "models/";
package models;
import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

enum PacketType {
  INIT_REQUEST = 0;
  INIT_RESPONSE = 1;
  GETATTR_REQUEST = 2;
  GETATTR_RESPONSE = 3;
}

message InitRequest { string project_name = 1; }

message InitResponse { int32 error = 1; }

message GetattrRequest { string path = 1; }

message GetattrResponse {
  int32 error = 1;
  message stat {
    int32 mode = 1;
    int64 size = 2;
    google.protobuf.Timestamp mtime = 3;
  }
  stat stbuf = 2;
  int32 result = 3;
}
